src/controllers/planExercise/index.ts

```diff
@@ -0,0 +1,126 @@
+import { Request, Response } from 'express';
+import PlanExerciseService from '../../services/planExercise';
+import {
+  CreatePlanExerciseRequest,
+  FindPlanExerciseRequest,
+  PatchPlanExerciseRequest
+} from './types';
+import { handlePrismaError } from '../../lib/handle-prisma-error';
+import ExerciseService from '../../services/exercise';
+import TrainingPlanService from '../../services/trainingPlan';
+
+export default class PlanExerciseController {
+  static async getAll(_: Request, response: Response) {
+    try {
+      const planExercises = await PlanExerciseService.getAll();
+      return response.success({ data: planExercises });
+    } catch (err) {
+      return response.error(handlePrismaError(err));
+    }
+  }
+
+  static async find(request: Request, response: Response) {
+    const { id } = request.params as any as FindPlanExerciseRequest;
+    if (!id) {
+      return response.badrequest({ errors: { id: 'ID is required' } });
+    }
+
+    const planExercise = await PlanExerciseService.find({ id });
+
+    if (!planExercise) {
+      return response.notfound();
+    }
+
+    return response.success({ data: planExercise });
+  }
+
+  static async create(request: Request, response: Response) {
+    const body = request.body as any as CreatePlanExerciseRequest;
+
+    if (!body.plan_id) {
+      return response.badrequest({ errors: { name: 'Plan is required' } });
+    }
+
+    if (!body.exercise_id) {
+      return response.badrequest({ errors: { name: 'Exercise is required' } });
+    }
+
+    if (!body.repetitions) {
+      return response.badrequest({ errors: { name: 'Repetition is required' } });
+    }
+
+    if (!body.rest_between_sets) {
+      return response.badrequest({ errors: { name: 'Rest between sets is required' } });
+    }
+
+    if (!body.sets) {
+      return response.badrequest({ errors: { name: 'Sets are required' } });
+    }
+
+    if (!body.day) {
+      return response.badrequest({ errors: { name: 'Day is required' } });
+    }
+
+    try {
+      const plan = await TrainingPlanService.find({ id: body.plan_id });
+      const foundExercise = await ExerciseService.find({ id: body.exercise_id });
+
+      if (!plan) {
+        return response.badrequest({ errors: { plan_id: 'Invalid Plan ID provided' } });
+      }
+
+      if (!foundExercise) {
+        return response.badrequest({ errors: { machine_id: 'Invalid Exercise ID provided' } });
+      }
```

Comment:
do a find in database 1 by 1. no need to try to find the exercise if plan doesn't exist.

also rename `foundExercise` to `exercise`